website/static/js/wikiDiff.js

```diff
@@ -0,0 +1,38 @@
+var jsdiff = require('jsdiff');
+
+// Taken from diff.js
+function removeEmpty(array) {
+    var ret = [];
+    for (var i = 0; i < array.length; i++) {
+        if (array[i]) {
+            ret.push(array[i]);
+        }
+    }
+    return ret;
+}
+
+// Custom differ that includes newlines and sentence breaks
+var wikiDiff =  new jsdiff.Diff();
+wikiDiff.tokenize = function (value) {
+    return removeEmpty(value.split(/(\S.+?[.!?\n])(?=\s+|$)/));
+};
+
+var diff = function(beforeText, afterText) {
+    var diffList = wikiDiff.diff(beforeText, afterText);
+    var fragment = document.createDocumentFragment();
+
+    diffList.forEach(function(part) {
```

Comment:
I don't know that it will be in this case. The comment at the bottom suggests, "If you can keep the contents of the for loop simple, then consider it. but if the contents get complex such that it would be better implemented in a function, then the forEach loop probably runs about equal and is certainly more readable." And the speed for this is going to be in touching the dom rather than iterating through the loop. With more items and less interior code, I am sure it would be, but in this case, I have my doubts.